doc/src/atomvm-tooling.md

```diff
@@ -0,0 +1,453 @@
+
+    shell$ atomvm _build/default/lib/hello_world.avm
+    Hello World
+    Return value: ok
+
+For instructions about how to install AtomVM on the `generic_unix` platform, see the [Getting Started Guide](getting-started-guide.md)
+
+### Flashing your application
+
+The [`atomvm_rebar3_plugin`](https://atomvm.github.io/atomvm_rebar3_plugin) supports flash targets for various device types.  These targets are described in more detail below.
+
+#### ESP32
+
+To flash AtomVM AVM file to an ESP32 device, use the `esp32_flash` target in the `atomvm` namespace.  Users will typically specify the device port and baud rate as command-line options to this target.
+
+> Note. In order to use the `esp32_flash` target, you will need to install the [`esptool`](https://github.com/espressif/esptool) program.
+
+For example:
+
+    shell$ rebar3 atomvm esp32_flash --port /dev/ttyUSB0 --baud 921600
+    ...
+    ===> esptool.py --chip auto --port /dev/ttyUSB0 --baud 921600 --before default_reset --after hard_reset write_flash -u --flash_mode keep --flash_freq keep --flash_size detect 0x210000 /path/to/atomvm_examples/erlang/hello_world/_build/default/lib/hello_world.avm
+
+See the [`atomvm_rebar3_plugin`](https://atomvm.github.io/atomvm_rebar3_plugin) page for more detailed instructions about how to use the `esp32_flash` target.
+
+You can now use a serial console program such as [minicom](https://en.wikipedia.org/wiki/Minicom) or [screen](https://en.wikipedia.org/wiki/GNU_Screen) to view console output from a device.
+
+For example:
+
+        ###########################################################
+
+           ###    ########  #######  ##     ## ##     ## ##     ##
+          ## ##      ##    ##     ## ###   ### ##     ## ###   ###
+         ##   ##     ##    ##     ## #### #### ##     ## #### ####
+        ##     ##    ##    ##     ## ## ### ## ##     ## ## ### ##
+        #########    ##    ##     ## ##     ##  ##   ##  ##     ##
+        ##     ##    ##    ##     ## ##     ##   ## ##   ##     ##
+        ##     ##    ##     #######  ##     ##    ###    ##     ##
+
+        ###########################################################
+
+    I (852) AtomVM: Starting AtomVM revision 0.6.0-alpha.1
+    I (862) sys: Loaded BEAM partition boot.avm at address 0x1d0000 (size=262144 bytes)
+    I (882) network_driver: Initialized network interface
+    I (882) network_driver: Created default event loop
+    I (902) AtomVM: Found startup beam esp32init.beam
+    I (922) AtomVM: Starting esp32init.beam...
+    ---
+    AtomVM init.
+    I (932) sys: Loaded BEAM partition main.avm at address 0x210000 (size=1048576 bytes)
+    Starting application...
+    Hello World
+    AtomVM finished with return value: ok
+    I (972) AtomVM: AtomVM application terminated.  Going to sleep forever ...
+
+#### STM32
+
+To flash AtomVM AVM file to an STM32 device, use the `stm32_flash` target in the `atomvm` namespace.
```

Comment:
We should add a note that the packed avm must also include atomvmlib.avm (and should be pruned) before flashing. The command to prepare would look like:

`rebar3 atomvm packbeam -p -e /path/to/atomvmlib.avm`

But I can add that in a future PR if you would prefer.

Reply:
Corrected, but changed the instructions slightly to use the new `rebar.config` options (so you don't have to run the packbeam every time).